File: pset2/caesar/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <ctype.h>

int main(int argc, string argv[])
{
    string plaintext;
    
    if(argc != 2)
    {
        printf("%s k\n",argv[0]);
        return 1;
    }
    
    int k = atoi(argv[1]);
    
    printf("plaintext: ");
    plaintext = get_string();
    
    printf("ciphertext: ");
    for(int i = 0, n = strlen(plaintext); i< n; i++)
    {
        if(isalpha(plaintext[i]) && isupper(plaintext[i]))
        {
            printf("%c", ((plaintext[i] - 65 + k) % 26) +65);
        }
        else if((plaintext[i]) && islower(plaintext[i]))
        {
            printf("%c",((plaintext[i] - 97 + k) % 26) +97);
        }
        else
        {
            printf("%c",plaintext[i]);
        }
    }
    printf("\n");
    
    return 0;
}

File: pset6/caesar.py
import cs50
import sys

if len(sys.argv) != 2:
    print("{} k".format(sys.argv[0]))
    exit(1)
    
k = int(sys.argv[1])
print(k)

print("plaintext: ", end="")
plaintext = cs50.get_string()

print("ciphertext: ", end="")
for i in plaintext:
    if i.isalpha() and i.isupper():
        print(chr((ord(i) - 65 + k) % 26 + 65), end="")
    elif i.isalpha() and i.islower():
        print(chr((ord(i) - 97 + k) % 26 + 97), end="")
    else:
        print(i, end="")
        
print()
exit(0)